src/core/knn.js

```diff
@@ -0,0 +1,131 @@
+'use strict';
+
+const { recursiveToJSON, checkType } = require('./util');
+const Query = require('./query');
+
+/**
+ * Class representing a k-Nearest Neighbors (k-NN) query.
+ * This class extends the Query class to support the specifics of k-NN search, including setting up the field,
+ * query vector, number of neighbors (k), and number of candidates.
```

Comment:
Could you add at least 1 example similar to [elastic-builder@v2.28.0`/src/queries/full-text-queries/combined-fields-query.js#L27-L28`](https://github.com/sudo-suhas/elastic-builder/blob/v2.28.0/src/queries/full-text-queries/combined-fields-query.js#L27-L28)?